mmv1/templates/terraform/pre_update/alloydb_cluster.go.erb

```diff
@@ -1,3 +1,38 @@
+// Implementation for cluster upgrade
+if d.HasChange("database_version") && !tpgresource.IsEmptyValue(reflect.ValueOf(d.Get("database_version"))) {
+    upgradeUrl := strings.Split(url, "?updateMask")[0] + ":upgrade"
+    patchObj := make(map[string]interface{})
+    patchObj["version"] = obj["databaseVersion"]
+    if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+        billingProject = bp
+    }
+    upgradeUrl = strings.Replace(upgradeUrl, "/v1/", "/v1alpha/", 1)
+
+    upgradeClusterTimeout := 72 * time.Hour
```

Comment:
Is the timeout 72 hour? It is too long for an operation's wait time.

Reply:
Yes. Actually, AlloyDB cluster upgrade is an LRO which can take up to 3 days in worst case.

Are there any repercussions of keeping this long a wait time? Will the TF code break or will there be any issues that can come due to this?

Reply:
Confirmed with the team, we have 58 hours of wait time out on this LRO.

Reply:
Thanks for the information. Do you know how long will it cost on average?

Reply:
There are no average numbers as we haven't done the profiling on the AlloyDB fleet.

The upgrades in our shadow testing takes around 1 to 3 hours for initial numbers.

The total time depends on the size of the database and the number of database objects.

Reply:
A client side field defaulted to true can be used to indicated waiting for the LRO or not.
Doc: https://googlecloudplatform.github.io/magic-modules/develop/client-side-fields/

Example: [skip_await_rollout](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/osconfig/resource_os_config_os_policy_assignment.go#L1053)

Reply:
Thanks, Zhenhua for helping me with this flag! Can you help me understand the there a problem with a long wait time?

Reply:
The purpose of this flag is to allow the users to do other tasks with Terraform instead of having to wait for long time. Without the flag, the users have to wait until the update is complete and cannot do anything else.

Reply:
@GauravJain21 , do you have any questions about this flag? Thanks.